service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/RespondentController.java

```diff
@@ -30,55 +31,61 @@
 @RequestMapping("/callback/enter-respondents")
 public class RespondentController {
 
-    private final MapperService mapper;
+    private final ObjectMapper mapper;
     private final RespondentService respondentService;
 
     @Autowired
-    public RespondentController(MapperService mapper,
-                                RespondentService respondentService) {
+    public RespondentController(ObjectMapper mapper, RespondentService respondentService) {
         this.mapper = mapper;
         this.respondentService = respondentService;
     }
 
     @PostMapping("/about-to-start")
     public AboutToStartOrSubmitCallbackResponse handleAboutToStart(@RequestBody CallbackRequest callbackrequest) {
         CaseDetails caseDetails = callbackrequest.getCaseDetails();
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
 
-        return respondentService.setMigratedValue(caseDetails);
+        caseDetails.getData().put("respondentsMigrated", respondentService.setMigratedValue(caseData));
+        caseDetails.getData().put("respondents1", respondentService.expandRespondentCollection(caseData));
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(caseDetails.getData())
+            .build();
     }
 
     @PostMapping("/mid-event")
     public AboutToStartOrSubmitCallbackResponse handleMidEvent(@RequestBody CallbackRequest callbackrequest) {
         CaseDetails caseDetails = callbackrequest.getCaseDetails();
 
         return AboutToStartOrSubmitCallbackResponse.builder()
-            .data(callbackrequest.getCaseDetails().getData())
+            .data(caseDetails.getData())
             .errors(validate(caseDetails))
             .build();
     }
 
     @PostMapping("/about-to-submit")
     public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody CallbackRequest callbackRequest) {
         CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+
+        caseDetails.getData().put("respondents1", respondentService.addHiddenValues(caseData));
 
-        return respondentService.addHiddenValues(caseDetails);
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(caseDetails.getData())
+            .build();
     }
 
-    @SuppressWarnings("unchecked")
     private List<String> validate(CaseDetails caseDetails) {
         ImmutableList.Builder<String> errors = ImmutableList.builder();
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
```

Comment:
Is this needed? Can we not replace the else if below with just` else if (caseData.getRespondents() != null)`

Reply:
Yes, converts my data into a more usable format rather than needing to operate on maps